cf_remote/ssh.py

```diff
@@ -30,6 +27,33 @@ def put(self, src):
             shutil.copy(src, dst)
 
 
+class Connection:
+    def __init__(self, host, user, connect_kwargs=None):
+        self.ssh_host = host
+        self.ssh_user = user
+        self._connect_kwargs = connect_kwargs
+
+    def run(self, command, hide=False):
+        extra_ssh_args = []
+        if "key_filename" in self._connect_kwargs:
+            extra_ssh_args.extend(["-i", self._connect_kwargs["key_filename"]])
+
+        ahost = aramid.Host(self.ssh_host, self.ssh_user, extra_ssh_args)
+        results = aramid.execute([ahost], command, echo=(not hide))
+        return results[ahost][0]
+
+    def put(self, src):
+        ahost = aramid.Host(self.ssh_host, self.ssh_user)
+        results = aramid.put([ahost], src)
+        return results[ahost][0].retcode
+
+    def __enter__(self, *args, **kwargs):
+        return self
+
+    def __exit__(self, *args, **kwargs):
+        pass
```

Comment:
Why use `__enter__` and `__exit__` when not managing any resources?

Reply:
Because I wanted to make minimal changes and we were using the returned value of `connect()` in a `with` clause.